markdowns/19_Text_to_Image_search_pipeline_with_MultiModal_Retriever.md

```diff
@@ -0,0 +1,190 @@
+
+## Preparing the Colab Environment
+
+- [Enable GPU Runtime in GPU](https://docs.haystack.deepset.ai/v5.2-unstable/docs/enable-gpu-runtime-in-colab)
+- [Check if GPU is Enabled](https://docs.haystack.deepset.ai/v5.2-unstable/docs/check-if-gpu-is-enabled)
+- [Set logging level to INFO](https://docs.haystack.deepset.ai/v5.2-unstable/docs/set-the-logging-level)
+
+
+```bash
+%%bash
+
+pip install --upgrade pip
+pip install farm-haystack[colab]
+```
+
+# Initializing the DocumentStore
+
+A DocumentStore contains Documents, which in this case are references to the images that Haystack will compare with your query. Here we are using the InMemoryDocumentStore since it requires no external dependencies. To learn more about the DocumentStore and the different types of external databases that we support, see [DocumentStore](https://docs.haystack.deepset.ai/docs/document_store).
+
+
+```python
+from haystack.document_stores import InMemoryDocumentStore
+
+# Here we initialize the DocumentStore with 512 dims 
+# as we will use it to store 512 dim image embeddings 
+document_store = InMemoryDocumentStore(embedding_dim=512)
+```
+
+# Downloading Data
+
+Download 18 sample images of different animals from . You can find them in data/tutorial19/spirit-animals/ as a set of .jpg files.
+
+
+```python
+from haystack.utils import fetch_archive_from_http
+
+doc_dir = "data/tutorial19"
+
+fetch_archive_from_http(
+    url="https://s3.eu-central-1.amazonaws.com/deepset.ai-farm-qa/datasets/documents/spirit-animals.zip",
+    output_dir=doc_dir
+)
+```
+
+Add the images you just downloaded into Haystack Document objects and write them into the DocumentStore.
+
+
+```python
+import os
+
+from haystack import Document
+
+images = [
+    Document(content=f"./{doc_dir}/spirit-animals/{filename}", content_type="image")
+    for filename in os.listdir(f"./{doc_dir}/spirit-animals/")
+]
+
+document_store.write_documents(images)
+```
+
+# Initializing the Retriever
+
+Retrievers sift through all the images and return only those that are relevant based on the input query. Here we are using the [OpenAI CLIP model](https://github.com/openai/CLIP/blob/main/model-card.md) to embed images. For more details on supported modalities, see [MultiModalRetriever](https://docs.haystack.deepset.ai/docs/retriever#multimodal-retrieval).
+
+
+```python
+from haystack.nodes.retriever.multimodal import MultiModalRetriever
+
+retriever_text_to_image = MultiModalRetriever(
+    document_store=document_store,
+    query_embedding_model = "sentence-transformers/clip-ViT-B-32",
+    query_type="text",
+    document_embedding_models = {"image": "sentence-transformers/clip-ViT-B-32"}
+)
+
+document_store.update_embeddings(retriever=retriever_text_to_image)
+```
+
+# Creating the MultiModal search Pipeline
+
+We are populating a Pipeline with a MultiModalRetriever node. This search pipeline queries the image database with text and returns the most relevant images.
```

Comment:
I think we've already said that before. Here, I'd say something like: Time to create the pipeline for our search! Define the pipeline components: